### DISCUSSION TITLE
Is it possible to direct render graph in client

### DISCUSSION THREAD

**Discussion by wss29** [eclipse-glsp/glsp#1405] [Q&A]
Hi,
    I want to render a graph in glsp, I do not need to add or remove nodes, just render the graph in glsp theia
    it seems that I do not need the glsp-server, because I get graph data in frontend, If I can render graph directly in client, It will be simple

**Comment by tortmayr** [comment]
Hi @wss29,
sounds like your usecase is not the right fit for GLSP.
If you  just want to render a Diagram and don`t need/want editing capabilities the client-server approach of GLSP is an overkill.
For this usecase I would recommend to directly use [sprotty](https://sprotty.org/)
